### PR TITLE
Allow empty data writes for resources

### PR DESCRIPTION
**Pull request by skpratt**
some types of resources have no data (like workload identity)

### CODE REVIEW

File: agent/grpc-external/services/resource/write_test.go [hunk 1]
```diff
@@ -46,10 +46,6 @@ func TestWrite_InputValidation(t *testing.T) {
 			artist.Id.Name = ""
 			return artist
 		},
-		"no data": func(artist, _ *pbresource.Resource) *pbresource.Resource {
-			artist.Data = nil
-			return artist
-		},
```
Comment (ishustava) [inline agent/grpc-external/services/resource/write_test.go:52]: should we add a test that empty data is allowed?

**Review by ishustava — APPROVED** [review]
Looks good but adding a comment about adding a test case to make sure we cover that empty data is allowed.

File: agent/grpc-external/services/resource/write_test.go (hunk 1 above)
Comment (skpratt) [reply agent/grpc-external/services/resource/write_test.go:52]: done
